### PR TITLE
Adjust data format for download all users

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

Adjust the format of data coming back a bit to make things easier for the front end.  Add a test.

## Security Considerations

N/A

### CODE REVIEW

**Review by A-Shumway42 — APPROVED** [review]
Looks good to me, just have one question for clarity.
Is it necessary to do `json.dumps()` before returning with `jsonify()`?

**Comment by terrazoon** [comment]
> Looks good to me, just have one question for clarity. Is it necessary to do `json.dumps()` before returning with `jsonify()`?

Well given that I intend to call jsonify() it is necessary to convert the list of dicts into a string, because jsonify() seems to expect one.  I might be able to skip the call to both and just return the original data, but I want to call jsonify() for consistency's sake.

**Comment by xlorepdarkhelm** [comment]
> > Looks good to me, just have one question for clarity. Is it necessary to do `json.dumps()` before returning with `jsonify()`?
> 
> Well given that I intend to call jsonify() it is necessary to convert the list of dicts into a string, because jsonify() seems to expect one. I might be able to skip the call to both and just return the original data, but I want to call jsonify() for consistency's sake.

I thought jsonify() actually converted everything over into strings.... kinda the point.

File: app/user/rest.py [hunk 1]
```diff
@@ -629,7 +629,18 @@ def get_all_users():
...
+    for user in users:
+        new_user = {
+            "name": user[0],
+            "email_address": user[1],
+            "mobile_number": user[2],
+            "service": user[3],
+        }
+        download_users.append(new_user)
+    print(f"download_users {download_users}")
+    dump = json.dumps(download_users)
```
Comment (xlorepdarkhelm) [inline app/user/rest.py:642]: Seems like this isn't necessary.

jsonify() is used to replace json.dumps(), I don't see the point of dumps first, seems error-prone.

https://www.geeksforgeeks.org/use-jsonify-instead-of-json-dumps-in-flask/

**Review by xlorepdarkhelm — CHANGES_REQUESTED** [review]
Has this been tried without the json.dumps() yet? Because it feels badong to do both here.

File: app/user/rest.py (hunk 1 above)
Comment (terrazoon) [reply app/user/rest.py:642]: I added the json.dumps() for a reason ...

Comment (ccostino) [reply app/user/rest.py]: It looks like a further adjustment was made here - @xlorepdarkhelm and @terrazoon, is everything in good shape now?  If yes, @xlorepdarkhelm could you please review again?  Thanks!

Comment (terrazoon) [reply app/user/rest.py]: It's fine as far as I'm concerned.

**Review by xlorepdarkhelm — APPROVED** [review]
LGTM

**Pull request merged**